docs/infra/pull-request-environments.md

```diff
@@ -0,0 +1,33 @@
+# Pull request environments
+
+A temporary environment is created for each pull request that stays up while the pull request is open. Use cases for the temporary pull request environment includes:
+
+- Allow other delivery stakeholders—including product managers, designers, and business owners—to review changes before being merged and deployed
+- Enable automated end-to-end tests on the pull request
+- Enable automated accessibility checks on the pull request
+
+## Lifecycle of pull request environments
+
+A pull request environment is created when a pull request is opened or reopened, and destroyed when the pull request is merged or closed. When new commits are pushed up to the pull request, the pull request environment is updated.
+
+## Isolate database migrations into separate pull requests
+
+Database migrations are not reflected in PR environments. In particular, PR environments shares the same database with the dev environment, so database migrations that exist in the pull request are not run on the database to avoid impacting the dev environment.
+
+Therefore, isolate database changes in their own pull request and merge that pull request first before opening pull requests with application changes that depend on those database changes. Note that it is still okay and encouraged to develop database and application changes together during local development.
+
+This guidance is not strict. It is still okay to combine database migrations and application changes in a single pull request. However, when doing so, note that the PR environment may not be fully functional if the application changes rely on the database migrations.
```

Comment:
Duplicate statement as in the previous paragraph?

Reply:
it's slightly redundant but not the same. the first paragraph says to isolate db migrations in PRs but it's ok to develop locally with the two at the same time. the second paragraph says that it's ok not to follow the previous paragraph guidance.